apps/backoffice-v2/src/common/hooks/useHomeLogic/useHomeLogic.ts

```diff
@@ -0,0 +1,21 @@
+import { useSearchParams } from 'react-router-dom';
+import { useZodSearchParams } from '@/common/hooks/useZodSearchParams/useZodSearchParams';
+import { HomeSearchSchema } from '@/pages/Home/home-search-schema';
+
+export const useHomeLogic = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [dateRange, setDateRange] = useZodSearchParams(HomeSearchSchema);
+
+  const handleDateRangeChange = (range: any) => {
```

Comment:
Instead of `any` you can do the following. Similarly you can access the type of the first argument of the `onChange` handler of the component.
```suggestion
  const handleDateRangeChange: ComponentProps<typeof DateRangePicker>['onChange'] = (range) => {
```